src/options.js

```diff
@@ -27,8 +27,14 @@ export function parseOptions(
 ) {
   flagProtection = flagProtection ? true : false;
   interpolation = interpolation ? true : false;
-  shell = isString(shell) ? shell : getDefaultShell({ env });
 
-  const shellName = getShellName({ shell }, { resolveExecutable });
+  let shellName = null;
+  if (isString(shell)) {
+    shellName = getShellName({ shell }, { resolveExecutable });
+  } else if (isTruthy(shell)) {
+    shell = getDefaultShell({ env });
+    shellName = getShellName({ shell }, { resolveExecutable });
```

Comment:
Ideally, the duplication of `shellName = getShellName({ shell }, { resolveExecutable });` is avoided (without introducing another smell).

Reply:
Resolved with 5a6fc52765880b65154f265e079031e381b20523